c++/src/kj/compat/http.c++

```diff
@@ -4077,6 +4556,28 @@ public:
     return getClient(parsed).openWebSocket(path, headersCopy);
   }
 
+  kj::Promise<ConnectResponse> connect(kj::StringPtr host, const HttpHeaders& headers) override {
```

Comment:
I think `connect()` may not be meaningful for `NetworkHttpClient`.

`NetworkHttpClient` is an `HttpClient` which connects directly to the server specified in the URL. I.e. it's used specifically when there is no proxy. The CONNECT method, meanwhile, only makes sense with proxies.

If anything, `NetworkHttpClient::connect()` should actually make a TCP connection to the given host, and not use HTTP at all. I am not sure, though, if that's actually useful to implement, vs. just always throwing an exception.